File: frontend/src/types/project.ts
export type ProjectStatus =
  | 'RECOMMENDED'
  | 'UNDER_REVIEW'
  | 'CLARIFICATION_REQUIRED'
  | 'SANCTIONED'
  | 'REJECTED'
  | 'ASSIGNED'
  | 'IN_PROGRESS'
  | 'VERIFICATION_PENDING'
  | 'VERIFIED'
  | 'COMPLETED'
  | 'CLOSED';

export type VerificationPriority = 'NORMAL' | 'ATTENTION' | 'HIGH_PRIORITY';

export interface Project {
  id: string;
  projectCode: string;
  workName: string;
  description?: string;
  allocationId?: string;
  mpName?: string;
  districtId: string;
  districtName?: string;
  constituencyId: string;
  constituencyName?: string;
  agencyId?: string;
  agencyName?: string;
  sector: string;
  estimatedCost: number;
  sanctionedCost?: number;
  actualExpenditure: number;
  physicalProgress: number;
  financialProgress: number;
  status: ProjectStatus;
  priority: VerificationPriority;
  trustScore: number;
  explanation?: string;
  latitude?: number;
  longitude?: number;
  provenance?: string;
  createdAt: string;
  updatedAt: string;
}


export interface AnomalyFinding {
  reasonCode: string;
  severity: VerificationPriority;
  explanation: string;
  affectedField?: string;
  recommendedAction: string;
}

export interface TrustAssessmentData {
  projectId: string;
  projectCode: string;
  trustScore: number;
  verificationAssessment: string;
  totalSignalsCount: number;
  findings: AnomalyFinding[];
  radarFactors: Record<string, string>;
  explanationHeadline: string;
  disclaimer: string;
}

File: backend/app/schemas/project.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import date, datetime

class ProjectBase(BaseModel):
    work_name: str = Field(..., min_length=3, description="Name of the proposed MPLADS work")
    description: Optional[str] = None
    district_id: str
    constituency_id: str
    sector: str
    estimated_cost: float = Field(..., ge=0, description="Estimated project cost in INR")
    mp_name: Optional[str] = None
    allocation_id: Optional[str] = None
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    expected_completion_date: Optional[date] = None

class ProjectCreate(ProjectBase):
    pass

from typing import Optional, List, Any

class ProjectResponse(ProjectBase):
    id: str
    project_code: str
    agency_id: Optional[str] = None
    sanctioned_cost: Optional[float] = 0.0
    actual_expenditure: float = 0.0
    physical_progress: float = 0.0
    financial_progress: float = 0.0
    status: str = "RECOMMENDED"
    priority: str = "NORMAL"
    trust_score: float = 100.0
    provenance: Optional[str] = "OFFICIAL"
    created_at: Optional[Any] = None
    updated_at: Optional[Any] = None

    class Config:
        from_attributes = True

class ProjectSanctionRequest(BaseModel):
    sanctioned_cost: float = Field(..., ge=0)
    remarks: Optional[str] = None

class ProjectAssignRequest(BaseModel):
    agency_id: str
    remarks: Optional[str] = None

class ProgressUpdateRequest(BaseModel):
    physical_progress: float = Field(..., ge=0, le=100)
    financial_progress: float = Field(..., ge=0, le=100)
    actual_expenditure: float = Field(..., ge=0)
    milestone: Optional[str] = None
    remarks: Optional[str] = None
